x-pack/plugins/code/server/lsp/workspace_handler.ts

```diff
@@ -382,4 +385,17 @@ export class WorkspaceHandler {
     const workspaceRelativePath = path.relative(this.workspacePath, workspaceRepo.workdir());
     this.revisionMap[workspaceRelativePath] = headCommit.sha().substring(0, 7);
   }
+
+  private async checkFile(workspaceRepo: Repository, file: string) {
+    const headCommit = await workspaceRepo.getHeadCommit();
```

Comment:
the `revision` from line 300 might not necessarily to be `HEAD` in here. do you think it makes more sense to pass around that revision in here instead of always using `HEAD`?

Reply:
Good catch.